### ISSUE TITLE
"TypeError: %d format: a number is required, not tuple" after a number of iteration

### ISSUE THREAD

**Issue by NTNguyen13** [scikit-optimize/scikit-optimize#715]
I define a model to be optimized by SkOpt `gp_minimize` according to this: http://slashtutorial.com/ai/tensorflow/19_hyper-parameters/

```
def create_model(cdepth1,cdepth2,cdepth3,cdepth4,cdepth5,dense1,csize1,csize2,csize3,csize4,csize5,max3,max4):
    input = Input(shape=(img_height, img_width, img_channels))
    x = Conv2D(cdepth1, csize1, padding='same')(input)
    x = ELU(alpha=1.0)(x)
    x = BatchNormalization(axis=channel_axis)(x)

    x = Conv2D(cdepth2, csize2, padding='same')(x)
    x = ELU(alpha=1.0)(x)
    x = BatchNormalization(axis=channel_axis)(x)
    x = MaxPooling2D(pool_size=(3,3))(x)

    x = Conv2D(cdepth3, csize3, padding='same')(x)
    x = ELU(alpha=1.0)(x)
    x = BatchNormalization(axis=channel_axis)(x)
    x = MaxPooling2D(pool_size=(3,3))(x)

    x = Conv2D(cdepth4, csize4, padding='same')(x)
    x = ELU(alpha=1.0)(x)
    x = BatchNormalization(axis=channel_axis)(x)
    x = MaxPooling2D(pool_size=max3)(x)

    x = Conv2D(cdepth5, csize5, padding='same')(x)
    x = ELU(alpha=1.0)(x)
    x = BatchNormalization(axis=channel_axis)(x)
    x = MaxPooling2D(pool_size=max4)(x)

    x = Flatten()(x)
    x = Dense(dense1)(x)
    x = ELU(alpha=1.0)(x)
    x = BatchNormalization(axis=channel_axis)(x)

    x = Dropout(0.5)(x)
    output = Dense(classes_num, activation='softmax')(x)
    model = Model(inputs=input, outputs=output)

    adam = optimizers.Adam(lr=0.001, beta_1=0.9, beta_2=0.999, epsilon=None, decay=0.00005, amsgrad=False)

    model.compile(loss='categorical_crossentropy',
                  optimizer=adam,
                  metrics=['accuracy'])

    return model
```

with those parameter space:

```
dimensions = [Categorical([16, 32, 64], name='cdepth1'),
             Categorical([16, 32, 64], name='cdepth2'),
             Categorical([32, 64, 128], name='cdepth3'),
             Categorical([64, 128, 256], name='cdepth4'),
             Categorical([64, 128, 256], name='cdepth5'),
             Categorical([500, 750, 1000], name='dense1'),
             Categorical([(3, 3), (4, 4), (5, 5)], name='csize1'),
             Categorical([(3, 3), (4, 4), (5, 5)], name='csize2'),
             Categorical([(3, 3), (4, 4), (5, 5)], name='csize3'),
             Categorical([(3, 3), (4, 4), (2, 2)], name='csize4'),
             Categorical([(3, 3), (4, 4), (2, 2)], name='csize5'),
             Categorical([(3, 3), (4, 4), (2, 2)], name='max3'),
             Categorical([(3, 3), (4, 4), (2, 2)], name='max4')]
```

after 7-8 successful iteration, this error suddenly appears:

```
Traceback (most recent call last):
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 2961, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-49c8f570da87>", line 175, in <module>
    x0=default_parameters)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/skopt/optimizer/gp.py", line 228, in gp_minimize
    callback=callback, n_jobs=n_jobs)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/skopt/optimizer/base.py", line 249, in base_minimize
    result = optimizer.tell(next_x, next_y)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/skopt/optimizer/optimizer.py", line 443, in tell
    return self._tell(x, y, fit=fit)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/skopt/optimizer/optimizer.py", line 486, in _tell
    est.fit(self.space.transform(self.Xi), self.yi)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/skopt/learning/gaussian_process/gpr.py", line 195, in fit
    super(GaussianProcessRegressor, self).fit(X, y)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/sklearn/gaussian_process/gpr.py", line 217, in fit
    self.kernel_.bounds))]
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/sklearn/gaussian_process/gpr.py", line 454, in _constrained_optimization
    fmin_l_bfgs_b(obj_func, initial_theta, bounds=bounds)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/scipy/optimize/lbfgsb.py", line 199, in fmin_l_bfgs_b
    **opts)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/scipy/optimize/lbfgsb.py", line 335, in _minimize_lbfgsb
    f, g = func_and_grad(x)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/scipy/optimize/lbfgsb.py", line 285, in func_and_grad
    f = fun(x, *args)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/scipy/optimize/optimize.py", line 293, in function_wrapper
    return function(*(wrapper_args + args))
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/scipy/optimize/optimize.py", line 63, in __call__
    fg = self.fun(x, *args)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/sklearn/gaussian_process/gpr.py", line 209, in obj_func
    theta, eval_gradient=True)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/sklearn/gaussian_process/gpr.py", line 412, in log_marginal_likelihood
    K, K_gradient = kernel(self.X_train_, eval_gradient=True)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/sklearn/gaussian_process/kernels.py", line 679, in __call__
    K1, K1_gradient = self.k1(X, Y, eval_gradient=True)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/sklearn/gaussian_process/kernels.py", line 754, in __call__
    K2, K2_gradient = self.k2(X, Y, eval_gradient=True)
  File "/home/cngc3/anaconda3/envs/tensorflow/lib/python3.6/site-packages/skopt/learning/gaussian_process/kernels.py", line 391, in __call__
    (X.shape, len(length_scale)))
TypeError: %d format: a number is required, not tuple
```

I'm quite confused about how this happen, since it can run smoothly before.
Thank you

**Comment by liebscher** [comment]
I recently experienced the same error and under similar circumstances. My solution was to replace Categoricals of integer values to an Integer distribution.

In your case:
`Categorical([3, 4, 5], name='csize1')` => `Integer(3, 5, name='csize1')`

I don't know if there's a solution to your other depth Categoricals. Perhaps change (decrease) the iteration size but sample over a similar range?

**Comment by jhell96** [comment]
I also had this error and tried to make a pull request, but realized it was already fixed in #650. 

It [looks like](https://pypi.org/project/scikit-optimize/#history) the last release of scikit-optimize was March 25, 2018. 

The library seems due for new one!
